### PR TITLE
Remove deprecated operators of BaseOperator

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

Remove the overloaded operators `*` (`__mul__`) for the `dot` method and `@` (`__matmul__`) for the `compose` of `BaseOperator`, that is deprecated as of version 0.17.0.

### Details and comments

label: `mod: quantum info`
milestone: 0.19

### CODE REVIEW

**Review by mtreinish — COMMENTED** [review]
Overall LGTM, the release not could just use some more details I left an inline suggestion on how to improve it, but since I did it via the github web editor I assume the whitespace and stuff is all wrong (github makes it very difficult to get correct).

**Review by jakelishman — COMMENTED** [review]
Are there any of the deprecation warning filters in `qiskit/test/base.py` that can be removed now that these have expired?

File: releasenotes/notes/remove-deprecated-ops-2fbd27abaee98c68.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
+  - |
+    Remove the overloaded operators `*` (`__mul__`) for the `dot` method and `@` (`__matmul__`) for
+    the `compose` of `BaseOperator`, that is deprecated as of version 0.17.0.
```
[the remark this answers is not in the document]
Comment (ikkoham) [reply releasenotes/notes/remove-deprecated-ops-2fbd27abaee98c68.yaml:5]: Thank you. I learned so much about writing the release note.

**Comment by ikkoham** [comment]
> Are there any of the deprecation warning filters in `qiskit/test/base.py` that can be removed now that these have expired?

Thanks. Yes. I can remove some. ce44e6b

**Review by jakelishman — APPROVED** [review]
I fixed a couple of issues in the release notes (for example the function `random_operator` doesn't exist!), but this all looks fine to me.  I'm particularly glad to be getting rid of some of these extra warning ignores from the test suite.

**Pull request merged**